src/linker/Linker/LinkerAttributesInformation.cs

```diff
@@ -77,5 +87,87 @@ static Attribute ProcessRequiresUnreferencedCodeAttribute (LinkContext context,
 				2028, MessageOrigin.TryGetOrigin (method, 0));
 			return null;
 		}
+
+		static DynamicDependency ProcessDynamicDependencyAttribute (LinkContext context, ICustomAttributeProvider provider, CustomAttribute customAttribute)
+		{
+			if (!(provider is MethodDefinition || provider is FieldDefinition))
+				return null;
+			var member = provider as IMemberDefinition;
+
+			if (!ShouldProcessDependencyAttribute (context, customAttribute))
+				return null;
+
+			var dynamicDependency = GetDynamicDependency (context, customAttribute);
+			if (dynamicDependency != null)
+				return dynamicDependency;
+
+			context.LogMessage (MessageContainer.CreateWarningMessage (context,
+				$"Invalid DynamicDependencyAttribute on '{member}'",
+				2034, MessageOrigin.TryGetOrigin (member)));
+			return null;
+		}
+
+		static DynamicDependency GetDynamicDependency (LinkContext context, CustomAttribute ca)
+		{
+			var args = ca.ConstructorArguments;
+			if (args.Count > 3)
+				return null;
+
+			// First argument is string or DynamicallyAccessedMemberTypes
+			string memberSignature = args[0].Value as string;
+			if (args.Count == 1)
+				return memberSignature == null ? null : new DynamicDependency (memberSignature);
+			DynamicallyAccessedMemberTypes? memberTypes = null;
+			if (memberSignature == null) {
+				var argType = args[0].Type;
+				if (!(argType.Namespace == "System.Diagnostics.CodeAnalysis" && argType.Name == "DynamicallyAccessedMemberTypes"))
+					return null;
+				try {
+					memberTypes = (DynamicallyAccessedMemberTypes) args[0].Value;
+				} catch (InvalidCastException) { }
+				if (memberTypes == null)
+					return null;
+			}
+
+			// Second argument is Type for ctors with two args, string for ctors with three args
+			if (args.Count == 2) {
+				if (!(args[1].Value is TypeReference type))
+					return null;
+				return memberSignature == null ? new DynamicDependency (memberTypes.Value, type) : new DynamicDependency (memberSignature, type);
+			}
+			Debug.Assert (args.Count == 3);
+			if (!(args[1].Value is string typeName))
+				return null;
+
+			// Third argument is assembly name
+			if (!(args[2].Value is string assemblyName))
+				return null;
+
+			var dynamicDependency = memberSignature == null ? new DynamicDependency (memberTypes.Value, typeName, assemblyName) : new DynamicDependency (memberSignature, typeName, assemblyName);
+			dynamicDependency.OriginalAttribute = ca;
+
+			return dynamicDependency;
+		}
+
+		public static bool ShouldProcessDependencyAttribute (LinkContext context, CustomAttribute ca)
+		{
+			if (ca.HasProperties && ca.Properties[0].Name == "Condition") {
```

Comment:
I'd probably not enable it for DynamicDependencyAttribute as it needs more thinking about what we want to do there